### PR TITLE
ADR Template Updates

### PR DESCRIPTION
**Pull request by Olshansk**
- Add a table of contents
- Add options for the deciders
- Add a background section
- Add required fields where appropriate
- Number the options being considered
- Add `Pros` and `Cons` sublists to the pros & cons section

### CODE REVIEW

File: ADRs/ADR_TEMPLATE.md [hunk 1]
```diff
@@ -1,72 +1,107 @@
-# [short title of solved problem and solution]
+# [short title of solved problem and solution] <!-- omit in toc -->
 
-* Status: [proposed | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)] <!-- optional -->
-* Deciders: [list everyone involved in the decision] <!-- optional -->
-* Date: [YYYY-MM-DD when the decision was last updated] <!-- optional -->
+- Status: [proposed | in review | rejected | accepted | deprecated | … | superseded by [ADR-0005](0005-example.md)]
```
Comment (bryanchriswhite) [inline ADRs/ADR_TEMPLATE.md:3]: Please review #36 when you get a chance, I've updated the statuses there as well as included the status flowchart in the README.

TLDR; these are the statuses I proposed:
```
draft | in review | changes needed | withdrawn | accepted | superseded
```

**Review by bryanchriswhite — APPROVED** [review]
:100: Wonderful improvements - thanks @Olshansk! :raised_hands:
